### PR TITLE
feat: new feature addition for chatbot

### PR DESCRIPTION
**Pull request by AceTheCreator**
**Description**

- Added multiple channels and messages functionality
- Added spec version question 
- Added docker to the base repo and required packages
- Added a well-detailed Readme

**Related issue(s)**

### CODE REVIEW

**Comment by derberg** [comment]
@AceTheCreator hey man, can you try editing [PR testing workflow](https://github.com/asyncapi/chatbot/blob/master/.github/workflows/if-nodejs-pr-testing.yml) to include
```
      - name: Set git to use LF
        run: |
          git config --global core.autocrlf false
          git config --global core.eol lf
```

to check if this can solve issue with line breaks on windows? it should be added before running linter

File: README.md [hunk 1]
```diff
@@ -1,20 +1,102 @@
...
+
+1. cd [chatbots](https://github.com/asyncapi/chatbot) to go into the project root
+2. yarn to install the dependencies
+
+### Running locally
+
+Follow this step to run the project with the live server
+
+1. cd into packages/test-client
+2. create a .env and add PORT=[http://167.71.46.87/](http://167.71.46.87/) API_KEY="reach out to the maintainers for an API key" to communicate with the live server.
```
Comment (derberg) [inline README.md:89]: please mention that somebody should rather get their own Wit account and train their own model, we should not share API Key 😅

**Review by derberg — COMMENTED** [review]
Please extend readme with instructions on how to quickly start project with docker compose

File: README.md (hunk 1 above)
Comment (derberg) [reply README.md:89]: also where is this IP coming from? isn't better to do `ACCESS_TOKEN=abs yarn start`? we should make it as easy to start as possible

Comment (AceTheCreator) [reply README.md:89]: Can you explain better what you mean?

**Comment by AceTheCreator** [comment]
> Please extend readme with instructions on how to quickly start project with docker compose

If they'll be using docker-compose they don't have to install projects dependencies right?

File: README.md (hunk 1 above)
Comment (AceTheCreator) [reply README.md:89]: > you can pass environment variables on runtime in terminal as `ACCESS_TOKEN=abs yarn start`, no need to edit .env file
> 


Okay

Comment (AceTheCreator) [reply README.md:89]: > also, the IP that you have in this instruction, what is it for? where is the IP coming from?

It's the IP address for the live server

Comment (derberg) [reply README.md:89]: oh, you mean you managed to deploy it? perfect! So the API_KEY is not for Wit.ai but for the deployed application?

also if it is IP then why it is called PORT? maybe ADDRESS?

**Comment by derberg** [comment]
> If they'll be using docker-compose they don't have to install projects dependencies right?

@AceTheCreator only if your Dockerfile is making sure they are installed. I see you have multistage docker files but not sure if they are actually installing deps first

**Comment by AceTheCreator** [comment]
> question why you have now 2 Docker files, one on root and one in a server dir?

@derberg that was a mistake

**Comment by AceTheCreator** [comment]
> @AceTheCreator separate Dockerfile for the client and docker-compose on the root to boot both easily, so whoever wanna have a try do not need Node environment. You do not need docker-compose for just one service at all.
> 
@derberg when someone runs docker-compose, I want the test-client to listen to the live server. But the chatbot server will listen to a port locally, and that server will not run cuz of the unavailable wit token.

**Comment by derberg** [comment]
@AceTheCreator when users want to use test-client with live server, they should just use yarn/npm or the docker image created for test-client with `docker run`, or build image on their own with `docker build` first. So users need to know how to start with your `yarn start` and `docker run`. Look for example here https://github.com/asyncapi/generator we explain how to use generator as npm package, and we also have a docker related section

docker componse is only for those that do not want to use live server, and want to start server and test-client locally.

**Comment by AceTheCreator** [comment]
> @AceTheCreator when users want to use test-client with live server, they should just use yarn/npm or the docker image created for test-client with `docker run`, or build image on their own with `docker build` first. So users need to know how to start with your `yarn start` and `docker run`. Look for example here https://github.com/asyncapi/generator we explain how to use generator as npm package, and we also have a docker related section
> 
> docker componse is only for those that do not want to use live server, and want to start server and test-client locally.

Then they'll have to cd into the test-client dir and run docker-compose or docker build

**Comment by derberg** [comment]
they have to `cd` only if they will use `yarn start` (unless you provide npm script to do it from the root like `npm run start:testClient`.
for docker-compose, you can have it on the root too.

for `docker run` they do not even have to clone the repo as image is anyway published on docker registry.

**Comment by AceTheCreator** [comment]
> for `docker run` they do not even have to clone the repo as image is anyway published on docker registry.
Yeah, I'm confused cuz when you run docker-compose from the root, it bootstraps both server and test-client.  And each package has its own docker image, and there's a docker-compose file in each package also. So I was thinking if there's a way to run docker-compose for a package(test-client for instance) from the root?

**Comment by derberg** [comment]
I think we are mixing few things here.

most important to clearly define what you need docker compose for -> when you run more than one service and do not want to call `docker run image1` and `docker run image2` but instead have one docker compose file where you compose 2 images together in one run, and call `docker-compose`

> Yeah, I'm confused cuz when you run docker-compose from the root, it bootstraps both server and test-client. And each package has its own docker image, and there's a docker-compose file in each package also.

so yeah, this is the intention, docker compose should start both, server and client

> So I was thinking if there's a way to run docker-compose for a package(test-client for instance) from the root

not docker compose as it doesn't make sense. Just to start test-client only, do `docker run --rm -it image_name` where `--rm` makes sure that once you stop, container is removed, and `-it` means you will have interactive session with container and after running it won't quit but you will be able to interact with it

**Comment by AceTheCreator** [comment]
@derberg after adding test-client, then when I run docker-compose up the test-client doesn't communicate with the server. Just like the Issue we had running it locally before

**Comment by derberg** [comment]
@AceTheCreator push what you have to the PR and I can take a look, otherwise hard to really say what the problem is

**Comment by derberg** [comment]
just make sure server really starts before client, or make sure client tries to reconnect as long as it needs

**Comment by AceTheCreator** [comment]
@derberg I've updated the PR. Please kindly check

**Review by derberg — COMMENTED** [review]
@AceTheCreator the problem I see is you are trying to embed 2 applications in the same container. You Dockerfile now builds an image with both, client and server. You need 2 images, a separate image for the server and a separate image for the client. Then in your docker-compose file you need to provide 2 services. Now you have `chatbot`, and you need `chatbot_server` and `chatbot_client`. So starting `docker-compose` will start both service, first should be server, then client (I mentioned the order in previous comment).

Summary what we need in Readme as final result:
- instruction on how to start the chatbot client only, and connect to hosted chatbot server - using `yarn`
- instruction on how to start the chatbot client only, and connect to hosted chatbot server - using `docker run`
- instruction on how to start locally both, server and client using own wit.ai API key - using `yarn`
- instruction on how to start locally both, server and client using own wit.ai API key - using `docker-compose`

does it make sense?

We can also focus in this PR on just making sure `yarn` instructions are in place and the whole docker thing can be delegated to a followup PR

**Comment by AceTheCreator** [comment]
> @AceTheCreator the problem I see is you are trying to embed 2 applications in the same container. You Dockerfile now builds an image with both, client and server. You need 2 images, a separate image for the server and a separate image for the client. Then in your docker-compose file you need to provide 2 services. Now you have `chatbot`, and you need `chatbot_server` and `chatbot_client`. So starting `docker-compose` will start both service, first should be server, then client (I mentioned the order in previous comment).
> 
> Summary what we need in Readme as final result:
> 
> * instruction on how to start the chatbot client only, and connect to hosted chatbot server - using `yarn`
> * instruction on how to start the chatbot client only, and connect to hosted chatbot server - using `docker run`
> * instruction on how to start locally both, server and client using own wit.ai API key - using `yarn`
> * instruction on how to start locally both, server and client using own wit.ai API key - using `docker-compose`
> 
> does it make sense?
> 
> We can also focus in this PR on just making sure `yarn` instructions are in place and the whole docker thing can be delegated to a followup PR

Okay

**Comment by AceTheCreator** [comment]
@derberg I've gat an update PR

**Review by derberg — APPROVED** [review]
don't forget to create a followup ticket for docker and related readme

:shipit:

**Pull request merged**